# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/tests/test_window_open_discarded_bc.html
# Commit: 6fd57b83bb52
# Full Hash: 6fd57b83bb52e9683bade8cbfb5f8ba6575ac6af
# Author: Kris Maglione <maglione.k@gmail.com>
# Date: 2020-02-07 03:55:51
# Regressor Bug: 1582832
# File Overlap Count: 3
# Description:
#   Bug 1582832: Part 1 - Make FrameLoader owner rather than DocShell responsible for discarding a BC. r=nika
#   
#   There are all sorts of lifecycle issues which arise from making DocShell
#   responsible for discarding BrowsingContexts. In this particular bug, we tend
#   to run into them in cases where we create a BrowsingContext for a FrameLoader,
# ==============================================================================

diff -r 28a9e4b075af -r 6fd57b83bb52 dom/ipc/tests/test_window_open_discarded_bc.html
--- a/dom/ipc/tests/test_window_open_discarded_bc.html	Thu Feb 06 18:35:10 2020 +0000
+++ b/dom/ipc/tests/test_window_open_discarded_bc.html	Thu Feb 06 19:07:56 2020 +0000
@@ -22,7 +22,13 @@
       // BrowsingContext to actually become discarded after we call it, to
       // make sure that the window provider actually has a discarded BC at the
       // end of its own nested event loop.
-      SpecialPowers.Services.tm.spinEventLoopUntil(() => !win.opener);
+      SpecialPowers.Services.tm.spinEventLoopUntil(() => {
+        try {
+          return !win.opener;
+        } catch (e) {
+          return false;
+        }
+      });
     }
   }
   SpecialPowers.addObserver(observer, TOPIC);